shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Val.hs

```diff
@@ -77,3 +84,52 @@ instance Val Coin where
   coin = id
   inject = id
   size _ = 1
+
+-- ============================================
+-- Constants needed to compute size and size-scaling operation
+
+-- address hash length is always same as Policy ID length
+addrHashLen :: Integer
+addrHashLen = 28
+
+smallArray :: Integer
+smallArray = 1
+
+hashLen :: Integer
+hashLen = 32
+
+uint :: Integer
+uint = 5
+
+hashObj :: Integer
+hashObj = 2 + hashLen
+
+addrHeader :: Integer
+addrHeader = 1
+
+address :: Integer
+address = 2 + addrHeader + 2 * addrHashLen
+
+-- input size
+inputSize :: Integer
+inputSize = smallArray + uint + hashObj
+
+-- size of output not including the Val (compute that part with vsize later)
+outputSizeWithoutVal :: Integer
+outputSizeWithoutVal = smallArray + address
+
+-- size of the UTxO entry (ie the space the scaled minUTxOValue deposit pays)
+utxoEntrySizeWithoutVal :: Integer
+utxoEntrySizeWithoutVal = inputSize + outputSizeWithoutVal
+
+-- This scaling function is right for UTxO, not EUTxO
+scaledMinDeposit :: (Val v) => v -> Coin -> Coin
+scaledMinDeposit v (Coin mv)
+  | inject (coin v) == v = Coin mv -- without non-Coin assets, scaled deposit should be exactly minUTxOValue
+  | otherwise = Coin $ fst $ quotRem (mv * (utxoEntrySizeWithoutVal + uint)) (utxoEntrySizeWithoutVal + size v) -- round down
+
+-- compare the outputs as Values (finitely supported functions)
```

Comment:
What is this comment doing? It's not associated with anything.

Reply:
removing